File: space-game/Data/Classes/RingData.vb
Imports Core.Classes
Imports Core.Extensions

Namespace Classes
    Public Class RingData
        Implements IEquatable(Of RingData)

        Public Property InnerRadius As Distance
        Public Property OuterRadius As Distance
        Public Property Textures As List(Of Texture)

        Public Shadows Function Equals(other As RingData) As Boolean Implements IEquatable(Of RingData).Equals
            If other IsNot Nothing Then
                Return InnerRadius.Equals(other.InnerRadius) AndAlso 
                    OuterRadius.Equals(other.OuterRadius) AndAlso 
                    Textures.IsEquivalent(other.Textures)
            End If
            Return False
        End Function
    End Class
End Namespace

File: space-game/Data/Classes/RingData.cs
using Core.Classes;
using Core.Extensions;
using System;
using System.Collections.Generic;

namespace Data.Classes
{
    public class RingData : IEquatable<RingData>
    {

        public Distance InnerRadius { get; set; }
        public Distance OuterRadius { get; set; }
        public List<Texture> Textures { get; set; }

        public bool Equals(RingData other)
        {
            if (other != null)
            {
                return InnerRadius.Equals(other.InnerRadius) && OuterRadius.Equals(other.OuterRadius) && Textures.IsEquivalent(other.Textures);
            }
            return false;
        }
    }
}
